blob/credential.go

```diff
@@ -20,8 +20,8 @@ import (
 // DeviceCredential contains all device state, including both public and private
 // parts of keys and secrets.
 type DeviceCredential struct {
-	Active bool
-	fdo.DeviceCredential
+	Active           bool
+	DeviceCredential fdo.DeviceCredential
```

Comment:
Before we change the marshaled form, which currently matches the non-normative [CDDL in the spec](https://fidoalliance.org/specs/FDO/fido-device-onboard-v1.0-ps-20210323/fido-device-onboard-v1.0-ps-20210323.html#DeviceCredential), let's try to figure out why marshaling embedded structs doesn't work on arm32.

Reply:
This is observed when we write the blob.DeviceCredentials to a file. 
To be precise, contents written to cred.bin after the DI completion contain empty/zero values for the fields inside embedded struct - fdo.DeviceCredential.

Reply:
upon further investigation, it is found that only last field of the embedded struct is marshalled and unmarshalled correctly. initial fields have zero values.

Reply:
Perhaps it is integers that work and not str/bstr? Thanks for the test! This should be enough to figure it out.

Reply:
Oh, no, it absolutely is order-based. Okay, so this is pretty bad. It means that there could be an ABI issue in the Go compiler. The reflect package does unsafe type casts to ABI types.

Reply:
So the cbor should be `84016474657374440c0d0e0f0f`, but the ARM version marshals to `84010f`, which is invalid. It can then also parse that invalid CBOR into the struct without error... This is immediately seeming less like a Go compiler issue and more like a bug in the CBOR package. Still odd that architecture affects it.

Reply:
Yes, the field order it calculates is wrong: 

```
2024/10/18 08:48:29 INFO fieldOrder indices="[[0] [1 2] [1 2] [1 2]]"
```

For x86 it is correct:

```
2024/10/18 08:49:07 INFO fieldOrder indices="[[0] [1 0] [1 1] [1 2]]"
```

Range variable issue?

Reply:
It is an issue with append(slice,int). reference copy is getting updated while iterating through embedded struct filed. unclear about why it happens only on arm32

Reply:
Updated the fix. Please review

Reply:
I had created a simple test for this usecase in cbor_test.go